### PR TITLE
chore: rename startup logs to agent logs

### PR DESCRIPTION
**Pull request by kylecarbs**
This also adds a `source` property to every agent log. It should allow us to group logs and display them nicer in the UI as they stream in.

### CODE REVIEW

File: codersdk/workspaceagents.go [hunk 1]
```diff
@@ -685,3 +685,11 @@ type AgentSubsystem string
 const (
 	AgentSubsystemEnvbox AgentSubsystem = "envbox"
 )
+
+type AgentLogSource string
+
+const (
+	AgentLogSourceStartup      AgentLogSource = "startup"
+	AgentLogSourceKubernetes   AgentLogSource = "kubernetes"
+	AgentLogSourceDevcontainer AgentLogSource = "devcontainer"
```
Comment (mafredri) [inline codersdk/workspaceagents.go:694]: Should these match the db enum?

**Review by mafredri — APPROVED** [review]
Love this change and that you went the extra mile with renaming code in the site as well!

Feel free to defer my suggestion about limit/source, as it's not critical for this refactor.

Didn't see any blockers so I'm approving. 👍🏻

File: codersdk/agentsdk/agentsdk.go [hunk 2]
```diff
@@ -567,20 +567,21 @@ func (c *Client) PostStartup(ctx context.Context, req PostStartupRequest) error
 	return nil
 }
 
-type StartupLog struct {
+type Log struct {
 	CreatedAt time.Time         `json:"created_at"`
 	Output    string            `json:"output"`
 	Level     codersdk.LogLevel `json:"level"`
+	Stage     string            `json:"stage"`
```
Comment (mafredri) [inline codersdk/agentsdk/agentsdk.go:574]: Should we match the db here and call this source?

File: codersdk/agentsdk/logs.go [hunk 3]
```diff
@@ -78,20 +78,20 @@ func (w *startupLogsWriter) Close() error {
 //
 // Neither Write nor Close is safe for concurrent use and must be used
 // by a single goroutine.
-func StartupLogsWriter(ctx context.Context, sender func(ctx context.Context, log ...StartupLog) error, level codersdk.LogLevel) io.WriteCloser {
+func StartupLogsWriter(ctx context.Context, sender func(ctx context.Context, log ...Log) error, level codersdk.LogLevel) io.WriteCloser {
```
Comment (mafredri) [inline codersdk/agentsdk/logs.go:81]: Should we rename this as well, `LogsWriter` and have it accept `source` in addition to level?

**Review by mafredri — COMMENTED** [review]
Two more things 😄

File: coderd/database/migrations/000141_workspace_agent_logs.up.sql [hunk 4]
```diff
@@ -0,0 +1,8 @@
+BEGIN;
+CREATE TYPE workspace_agent_log_source AS ENUM ('startup_script', 'kubernetes_logs', 'envbox', 'envbuilder', 'external');
+ALTER TABLE workspace_agent_startup_logs RENAME TO workspace_agent_logs;
+ALTER TABLE workspace_agent_logs ADD COLUMN source workspace_agent_log_source NOT NULL DEFAULT 'startup_script';
+ALTER TABLE workspace_agents RENAME COLUMN startup_logs_overflowed TO logs_overflowed;
```
[the remark this answers is not in the document]
Comment (kylecarbs) [reply coderd/database/migrations/000141_workspace_agent_logs.up.sql:5]: Ahh, that's a good point. I will change this!

File: coderd/database/migrations/000141_workspace_agent_logs.up.sql [hunk 5]
```diff
@@ -0,0 +1,8 @@
+BEGIN;
+CREATE TYPE workspace_agent_log_source AS ENUM ('startup_script', 'kubernetes_logs', 'envbox', 'envbuilder', 'external');
```
[the remark this answers is not in the document]
Comment (kylecarbs) [reply coderd/database/migrations/000141_workspace_agent_logs.up.sql:2]: Good point!

File: coderd/database/migrations/000141_workspace_agent_logs.down.sql [hunk 6]
```diff
@@ -0,0 +1,8 @@
+BEGIN;
+DROP TYPE workspace_agent_log_source;
+ALTER TABLE workspace_agent_logs RENAME TO workspace_agent_startup_logs;
+ALTER TABLE workspace_agent_startup_logs DROP COLUMN source;
```
[the remark this answers is not in the document]
Comment (kylecarbs) [reply coderd/database/migrations/000141_workspace_agent_logs.down.sql:4]: Yup, true!

File: codersdk/workspaceagents.go (hunk 1 above)
Comment (kylecarbs) [reply codersdk/workspaceagents.go:694]: Yup, they should. Good catch!

File: codersdk/agentsdk/agentsdk.go (hunk 2 above)
Comment (kylecarbs) [reply codersdk/agentsdk/agentsdk.go:574]: 100%

File: codersdk/agentsdk/logs.go (hunk 3 above)
Comment (kylecarbs) [reply codersdk/agentsdk/logs.go:81]: Yup, will fix!

File: coderd/database/migrations/000141_workspace_agent_logs.up.sql (hunk 4 above)
[the remark this answers is not in the document]
Comment (mafredri) [reply coderd/database/migrations/000141_workspace_agent_logs.up.sql:5]: I think a jsonb column, or a separate table (`agent_id`, `log_source`, `size`) (classic relational structure) would be fine. 👍🏻 In the relational variant we could move the overflow boolean to this new table as well.

[the remark this answers is not in the document]
Comment (kylecarbs) [reply coderd/database/migrations/000141_workspace_agent_logs.up.sql:5]: Going to do this after. Good idea!

**Pull request merged**